tests/code-check-roave-backward-compatibility/matrix.json

```diff
@@ -0,0 +1,10 @@
+{
+    "include": [
+        {
+            "name": "Backward Compatibility Check [@default, latest]",
+            "job": "{\"command\":\"roave-backward-compatibility-check check --from=\\\"1111222233334444aaaabbbbccccdddd\\\" --install-development-dependencies\",\"php\":\"@default\",\"extensions\":[],\"ini\":[],\"dependencies\":\"latest\",\"ignore_platform_reqs_8\":false,\"ignore_php_platform_requirement\":false,\"additional_composer_arguments\":[],\"before_script\":[]}",
```

Comment:
This uses `\"php\":\"@default\"`, but `roave/backward-compatibility-check` generally runs on very new versions.

I suppose this PHP version requiorement can be baked in the `roave-backward-compatibility-check` shell command: will check the companion PR to this.

Reply:
> This uses \"php\":\"@default\", but roave/backward-compatibility-check generally runs on very new versions.

Even tho there might be a version which might be compatible, the shell command will be installed while container is being created. Due to this, a specific version of `roave/backward-compatibility-check` is checked out. Pretty sure that if I check the latest version out, it does **not** work with PHP 5.6.
If it does, maybe you should allow PHP 5.6 within `composer.json`.

Reply:
I did not fully understand your last comment, so perhaps I'm misunderstanding what this does? :thinking:

Reply:
when the CI container is created and released, a default php Version is set. with the PR regarding roave bc check, I added composer.json to the root with a php Version compatible with all the Tools we do require, including roave bc check. This php Version is then set as the default PLUS runs composer install. 
It will remain as the default php Version. I do not have any clue in this container here, what default Version is executed in the container and thus, I am fine with having `@default` to ensure that whatever default version is set today and in 10 years, that it can run the bc check tool.
Since I do have to pass a php Version as it is required by the container and such, I rather give it a meaningful Name rather than sending 5.6 but the Tool is executed with 8.1 afterwards. 

the composer.lock in the CI container installs bc Tool which only allows php 8.1 and 8.2 and thus it is not compatible with any PHP Version or I totally miss understand what the php Version requirements are used for in that package.

So to not have to worry about Whatever php Version, I just Trust the container to run a php Version which can execute whatever bc Tool Version was installed during container build.

Reply:
Related to https://github.com/laminas/laminas-continuous-integration-action/pull/158#discussion_r1259450707

I'm fine with the current approach: let's try and roll with it :)